subscriber/podaac_data_downloader.py

```diff
@@ -273,6 +276,11 @@ def run(args=None):
             pa.process_file(process_cmd, output_path, args)
             logging.info(str(datetime.now()) + " SUCCESS: " + f)
             success_cnt = success_cnt + 1
+
+            #if limit is set and we're at or over it, stop downloading
+            if download_limit != -1 and success_cnt >= download_limit:
```

Comment:
so something funky was going on (never good). It kept complaining that i was referencing a variable before it was initialized- for some reason it saw me using download_limit as a local variable, not a 'global' one. i'll go back and clean that up because i do agree it's better that way.